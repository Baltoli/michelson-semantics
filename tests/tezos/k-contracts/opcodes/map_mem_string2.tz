$contract { parameter string;
storage (pair (map string nat) (option bool)) ;


code { UNPAIR;
       DIP { CAR; DUP };
       MEM; SOME; SWAP; PAIR; NIL operation; PAIR;}
 } ;
$param "foo" ;
$storage (Pair { Elt "foo" 0 } None) ;
$ExpectedOutput (Pair { Elt "foo" 0 } (Some True)) ;

