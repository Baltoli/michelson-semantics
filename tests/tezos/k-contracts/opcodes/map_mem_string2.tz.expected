<generatedTop>
  <k>
    .
  </k>
  <stack>
    Pair .List Pair #Map ( "foo" |-> 0 , string .AnnotationList , nat .AnnotationList ) Some true
  </stack>
  <paramtype>
    string .AnnotationList
  </paramtype>
  <storagetype>
    pair .AnnotationList map .AnnotationList string .AnnotationList nat .AnnotationList option .AnnotationList bool .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
